### PR TITLE
build(SF-1061): Replace re-start with custom release scripts

### PR DESCRIPTION
**Pull request by jmbtutor**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,49 +60,44 @@ jobs:
...
           <<: *node_cache
       - restore_cache:
-          key: docs_cache-{{ .Branch }}
+          key: build_cache-{{ .Branch }}
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
       - run:
-          name: Documentation
+          name: Setup Git user
```
Comment (Vic-Dev) [inline .circleci/config.yml]: Do we need this? Remove

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Modify the CHANGELOG.md
+
+Our CHANGELOG.md follows the [Keep a Changelog](https://keepachangelog.com) format. For every change add an Unreleased section by copying this format and adding it to the top of the CHANGELOG.md:
+
+```md
+## [Unreleased] [<release type>]
+### Changed
```
Comment (Vic-Dev) [inline CONTRIBUTING.md:19]: Added | Changed | Deprecated | Removed | Fixed | Security

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+## Modify the CHANGELOG.md
+
+Our CHANGELOG.md follows the [Keep a Changelog](https://keepachangelog.com) format. For every change add an Unreleased section by copying this format and adding it to the top of the CHANGELOG.md:
```
Comment (Vic-Dev) [inline CONTRIBUTING.md]: Describe how modifying the CHANGELOG triggers the release

File: scripts/publish.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+die() {
+  echo "ERROR:" "$@" >&2
+  exit 1
+}
+
+print_usage() {
+  cat <<EOF
+Usage: ${0##*/} [OPTIONS]
+       ${0##*/} -h
+Publishes the current package to npm.
```
Comment (Vic-Dev) [inline scripts/publish.sh]: `based on the version in package.json`

File: scripts/release.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (Vic-Dev) [inline scripts/release.sh:2]: Add shebang at the top

File: scripts/release.sh [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+git commit -m "Generate docs" ${CI:+'-m' "[ci skip]"} ./docs
+
+info "Determining the release type..."
+release_type="$(sed -n '/## \[Unreleased\] \[\(.*\)\]/ s//\1/p' CHANGELOG.md)"
+case "$release_type" in
+  major | minor | patch | premajor | preminor | prepatch | prerelease | from-git)
+    : # valid; do nothing
+    ;;
+  '')
+    die "Could not detect potential release in the CHANGELOG."
```
Comment (Vic-Dev) [inline scripts/release.sh]: Add exit code

File: scripts/release.sh [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    ;;
+  *)
+    die "Unsupported release type: ${release_type}"
```
Comment (Vic-Dev) [inline scripts/release.sh]: Add exit code

File: scripts/release.sh [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+/\[Unreleased\].*/ s//[${new_version#v}] - $(date +%F)/
+w
+q
+EOF
+
+info "Committing changes..."
+git commit -m "Release version ${new_version}" ${CI:+'-m' "[ci skip]"} package.json CHANGELOG.md
+
+info "Tagging commit..."
+sed -n '/## \[/,//p' CHANGELOG.md | sed -e '$d' -e 's/^##* *//' -e $'1a\\\n\\\n' |
```
Comment (Vic-Dev) [inline scripts/release.sh:96]: See if you can combine these seds

**Review by Vic-Dev — CHANGES_REQUESTED** [review]
Some minor things

File: scripts/release.sh (hunk 8 above)
Comment (JoeyDeol) [reply scripts/release.sh:96]: Decided to leave unchanged for clarity.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+- <Describe removals>
+
+### Fixed 
+- <Describe fixes>
+
+### Security
+- <Describe security fixes>
+```
+
+The release type corresponds to the arugement to `npm version`.
```
Comment (Vic-Dev) [inline CONTRIBUTING.md]: `argument`. Also mention specifically somewhere that the update to the CHANGELOG.md is what triggers the release

Comment (Vic-Dev) [inline CONTRIBUTING.md]: `argument`. Also mention specifically somewhere that the update to the CONTRIBUTING.md is what triggers the release

**Review by Vic-Dev — CHANGES_REQUESTED** [review]
One more thing

**Review by Vic-Dev — APPROVED** [review]

**Pull request merged**